### PR TITLE
Use --test-version correctly with kubetest2

### LINKED ISSUE #745: test_version not working for test/k8s-integration
/assign

### PR DESCRIPTION
**Pull request by mattcary**
/kind bug

Fixes #745

```release-note
None
```

### CODE REVIEW

File: test/k8s-integration/main.go [hunk 1]
```diff
@@ -657,6 +659,9 @@ func runTestsWithConfig(testParams *testParameters, testConfigArg, reportPrefix
 		kubeTest2Args = append(kubeTest2Args, fmt.Sprintf("--log_dir=%s", kubetestDumpDir))
 	}
 	kubeTest2Args = append(kubeTest2Args, "--")
+	if len(*testVersion) != 0 && *testVersion != "master" {
```
Comment (saikat-royc) [inline test/k8s-integration/main.go:662]: nit: Maybe we should move all the logic to prepare kubetest2 args under if *useKubeTest2{ ... }, to keep the code clean.

Comment (mattcary) [reply test/k8s-integration/main.go:662]: We're going to be tearing out kubetest; it's been helpful for debugging to be able to compare args.

Comment (saikat-royc) [reply test/k8s-integration/main.go:662]: Ack

**Comment by mattcary** [comment]
Now just seems to be problems with the fsgroup tests, possibly a race condition with the PVCs getting deleted.

/retest

**Pull request merged**